### PR TITLE
Provision serviceInstances defined in manifest

### PR DESCRIPTION
**Pull request by aidenkeating**
## Motivation

This provisions services that are defined in the manifest file. Currently it doesn't do anything further such as watching etc.

@david-martin Would you mind taking a look?

### CODE REVIEW

File: public/walkthroughs/my-custom-walkthrough/walkthrough.json [hunk 1]
```diff
@@ -1,6 +1,36 @@
 {
   "dependencies": {
     "repos": [],
-    "services": []
+    "serviceInstances": [{
+      "metadata": { "name": "fuse" },
+      "spec": {
```
Comment (david-martin) [inline public/walkthroughs/my-custom-walkthrough/walkthrough.json]: can probably leave a similar class as the example e.g. the nodejs messaging app.

**Comment by aidenkeating** [comment]
@david-martin Thanks. I'll work through the failing tests before merging

File: src/pages/tutorial/task/task.js [hunk 2]
```diff
@@ -42,44 +49,49 @@ class TaskPage extends React.Component {
...
-    if (currentProgress) {
-      this.setState({ verifications: currentProgress });
-    }
+    const currentUsername = localStorage.getItem('currentUserName');
+    const currentUserProgress = getThreadProgress(currentUsername);
+    updateWalkthroughProgress(currentUsername, currentUserProgress);
+
+    // const currentProgress = this.getStoredProgressForCurrentTask();
+    // if (currentProgress) {
+    //   this.setState({ verifications: currentProgress });
```
[the remark this answers is not in the document]
Comment (david-martin) [reply src/pages/tutorial/task/task.js]: leaving this here for now if you could comment on it later.

Merging with current changes

[the remark this answers is not in the document]
Comment (aidenkeating) [reply src/pages/tutorial/task/task.js]: It should be removed, I must have missed it

**Comment by aidenkeating** [comment]
@david-martin Apologies for only addressing these comments now. I removed the commented code in `task.js` and only provision the Node.js app now

**Pull request merged**